dwave/plugins/torch/models/losses/mmd.py

```diff
@@ -0,0 +1,183 @@
+# Copyright 2025 D-Wave
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from abc import abstractmethod
+from typing import Optional
+
+import torch
+import torch.nn as nn
+
+from dwave.plugins.torch.nn.modules.utils import store_config
+
+__all__ = ["Kernel", "RBFKernel", "mmd_loss"]
+
+
+class Kernel(nn.Module):
+    """
+    Base class for kernels.
+
+    Kernels are functions that compute a similarity measure between data points. Any ``Kernel``
+    subclass must implement the ``_kernel`` method, which computes the kernel matrix for a given
+    input multi-dimensional tensor with shape (n, f1, f2, ...), where n is the number of items
+    and f1, f2, ... are feature dimensions, so that the output is a tensor of shape (n, n)
+    containing the pairwise kernel values.
+    """
+
+    @abstractmethod
+    def _kernel(self, x: torch.Tensor) -> torch.Tensor:
+        """
+        Computes the kernel matrix for an input of shape (n, f1, f2, ...), whose shape is (n, n)
+        containing the pairwise kernel values.
+
+        Args:
+            x (torch.Tensor): A (n, f1, f2, ...) tensor.
+
+        Returns:
+            torch.Tensor: A (n, n) tensor.
+        """
+        pass
+
+    def forward(self, x: torch.Tensor, y: torch.Tensor) -> torch.Tensor:
+        """
+        Computes kernels for intra and inter set pairs in ``x`` and ``y``. In general, ``x`` and
+        ``y`` are (n_x, f1, f2, ...) and (n_y, f1, f2, ...) shaped tensors, and the output is a
+        (n_x + n_y, n_x + n_y) shaped tensor containing the pairwise kernel values.
+
+        Args:
+            x (torch.Tensor): A (n_x, f1, f2, ...) tensor.
+            y (torch.Tensor): A (n_y, f1, f2, ...) tensor.
+        Returns:
+            torch.Tensor: A (n_x + n_y, n_x + n_y) tensor.
+        """
+        assert x.shape[1:] == y.shape[1:], (
```

Comment:
Should be an if-check with an exception raised instead of an assert.

Reply:
Replaced it